Objects/obmalloc.c

```diff
@@ -1119,6 +1144,23 @@ _PyMem_AbandonDelayed(PyThreadState *tstate)
     assert(llist_empty(queue));  // the thread's queue is now empty
 }
 
+void
+_PyMem_ProcessAllDelayed(PyInterpreterState *interp)
+{
+    PyThreadState *tstate = PyInterpreterState_ThreadHead(interp);
```

Comment:
I think this is already handled by `process_delayed_frees` in `gc_free_threading.c`. FWIW, you would also need to lock `HEAD_LOCK(_PyRuntime);` to safely loop over thread states in the interpreter, even when we are in a stop-the-world pause.

Reply:
Hurm, weird, I'll need to look into what's going on here...  It does indeed look like `process_delayed_frees` should do the same thing, but the extra pass here is indeed preventing a crash.  It's not immediately obvious to me what's different.

Reply:
Oh, I just wasn't based on that revision yet, I'll remove all of this stuff...